### ISSUE TITLE
Digitalocean ssh key resources are always destroyed and re-created

### ISSUE THREAD

**Issue by cabbiepete** [hashicorp/terraform#8336]
Terraform 0.7.0

For a SSH Key:

```
resource "digitalocean_ssh_key" "april" {
  name = "april"
  public_key = "${file("<path>/april.pub")}"
}
```

Used by a droplet:
```
resource "digitalocean_droplet" "build" {
  image = "${var.standard_image}"
  name = "build"
  region = "${var.region}"
  size = "2gb"
  ipv6 = true
  private_networking = true
  ssh_keys = ["${digitalocean_ssh_key.april.fingerprint}"]
}
```

The ssh key is added correctly and hooked up to the droplet on creation fine.

However if I then run `plan` terraform wants to destroy and create the ssh key and also update the state of the droplet.

Currently this does not cause an incorrect setup in the end but it seems like terraform should really check for the existence of the ssh key as per the tfstate file and if it exists with the same fingerprint (and possibly name or id) do not destroy and recreate.

**Comment by stack72** [comment]
Hi @cabbiepete 

I believe this is the issue:

```
provider/digitalocean: trim whitespace from ssh key [GH-8173] 
```

This was released in 0.7.1 last night

Paul

**Comment by stack72** [comment]
ping @cabbiepete :)

**Comment by cabbiepete** [comment]
@stack72 confirmed fixed in 0.7.1
